pulsar-broker/src/main/java/org/apache/pulsar/broker/s3offload/DataBlockHeader.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.s3offload;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.bookkeeper.common.annotation.InterfaceStability.Unstable;
+import org.apache.pulsar.broker.s3offload.impl.DataBlockHeaderImpl;
+
+/**
+ * The data block header in code storage for each data block
+ *
+ * Currently, It is in format:
+ * [ magic_word -- int ][ block_len -- int ][ first_entry_id  -- long]
+ *
+ * with the size: 4 + 4 + 8 = 16 Bytes
+ */
+@Unstable
+public interface DataBlockHeader {
+    /**
+     * Get Magic Word for data block.
+     * It is a sequence of bytes used to identify the start of a block.
+     */
+    int getBlockMagicWord();
+
+    /**
+     * Get the length of the block in bytes, including the header.
+     */
+    int getBlockLength();
+
+    /**
+     * Get the message entry Id for the first message that stored in this data block.
+     */
+    long getFirstEntryId();
+
+    /**
+     * Get the size of this DataBlockHeader.
+     */
+    int getHeaderSize();
```

Comment:
How is header size useful to a user of the DataBlockHeader? I think this should return what getDataStartOffset is returning below. The contents of the header should be opaque to the user of the interface.

Reply:
Thanks. Seems this is the key for some of the comments below. If user get the actual size, user could ignore the paddings, InputStream passed into 'fromStream', and returned by 'toStream' could ignore the paddings.  
will change this to what you suggested.

Reply:
I don't think getHeaderSize() is needed anymore.